File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;
class Solution {
    public static int solution(String s) {
        int answer=0;
        int strLen = s.length();

        for(int i=0; i<strLen; i++){
            answer += cal(s, i, strLen);
        }

        return answer;
    }

    private static int cal(String s, int strIdx, int strLen){

        Stack<Character> st = new Stack<>();

        for(int i=strIdx; i<strIdx+strLen; i++){
            int idx = i%strLen;
            char ch = s.charAt(idx);

            if(ch == '(' || ch == '{' || ch =='['){
                st.push(ch);
            }else if(ch == ')' || ch == '}' || ch ==']'){
                if(st.empty()){
                    return 0;
                }else if((st.peek() != '(' && ch == ')') || (st.peek() != '{' && ch == '}') || (st.peek() != '[' && ch == ']')){
                    return 0;
                }else{
                    st.pop();
                }
            }
        }
        if(!st.empty()) return 0;
        return 1;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    cnt = 0
    num = len(s)
    s+=s
    for i in range(num):
        st = []

        for k in range(i, i+num):
            if len(st) == 0:
                st.append(s[k])
            else:
                if st[-1] == "(" and s[k] == ")":
                    st.pop()
                elif st[-1] == "{" and s[k] == "}":
                    st.pop()
                elif st[-1] == "[" and s[k] == "]":
                    st.pop()
                else:
                    st.append(s[k])
        
        if len(st) == 0:
            cnt += 1
    return cnt